tests/models/test_changestar.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import itertools
+
+import pytest
+import torch
+from torch.nn.modules import (
+    BatchNorm2d,
+    Conv2d,
+    Identity,
+    MaxPool2d,
+    Module,
+    ModuleList,
+    ReLU,
+    Sequential,
+    Sigmoid,
+    UpsamplingBilinear2d,
+)
+
+from torchgeo.models import ChangeMixin, ChangeStar, ChangeStarFarSeg
+
+Module.__module__ = "torch.nn"
+ModuleList.__module__ = "nn.ModuleList"
+Sequential.__module__ = "nn.Sequential"
+Conv2d.__module__ = "nn.Conv2d"
+BatchNorm2d.__module__ = "nn.BatchNorm2d"
+ReLU.__module__ = "nn.ReLU"
+UpsamplingBilinear2d.__module__ = "nn.UpsamplingBilinear2d"
+Sigmoid.__module__ = "nn.Sigmoid"
+Identity.__module__ = "nn.Identity"
+MaxPool2d.__module__ = "nn.MaxPool2d"
+
+BACKBONE = ["resnet18", "resnet34", "resnet50", "resnet101", "anynet"]
+IN_CHANNELS = [64, 128]
+INNNR_CHANNELS = [16, 32, 64]
+NC = [1, 2, 4]
+SF = [4.0, 8.0, 1.0]
+
+
+class TestChangeStar:
+    @torch.no_grad()  # type: ignore[misc]
+    def test_changestar_farseg_classes(self) -> None:
+        model = ChangeStarFarSeg(
+            classes=4, backbone="resnet50", backbone_pretrained=False
+        )
+        x = torch.randn(2, 2, 3, 128, 128)
+        y = model(x)
+
+        assert y["bi_seg_logit"].shape[2] == 4
+
+    @torch.no_grad()  # type: ignore[misc]
+    def test_changestar_farseg_output_size(self) -> None:
+        model = ChangeStarFarSeg(
+            classes=4, backbone="resnet50", backbone_pretrained=False
+        )
+        model.eval()
+        x = torch.randn(2, 2, 3, 128, 128)
+        y = model(x)
+
+        assert y["bi_seg_logit"].shape[3] == 128 and y["bi_seg_logit"].shape[4] == 128
+        assert y["change_prob"].shape[2] == 128 and y["change_prob"].shape[3] == 128
+
+        model.train()
+        y = model(x)
+
+        assert y["bi_seg_logit"].shape[3] == 128 and y["bi_seg_logit"].shape[4] == 128
+        assert y["bi_change_logit"][0].shape[2] == 128
+        assert y["bi_change_logit"][0].shape[3] == 128
+        assert y["bi_change_logit"][1].shape[2] == 128
+        assert y["bi_change_logit"][1].shape[3] == 128
+
+    @torch.no_grad()  # type: ignore[misc]
+    @pytest.mark.parametrize("backbone", BACKBONE)
+    def test_changestar_farseg_backbone(self, backbone: str) -> None:
+        if backbone == "anynet":
+            match = "unknown backbone: anynet."
+            with pytest.raises(ValueError, match=match):
+                ChangeStarFarSeg(
+                    classes=4, backbone="anynet", backbone_pretrained=False
+                )
+        else:
+            ChangeStarFarSeg(classes=4, backbone=backbone, backbone_pretrained=False)
```

Comment:
I think it would be simpler to split this into 2 tests, one that tests all valid args, and one that tests an invalid arg.